src/redux/reducers/eth.js

```diff
@@ -0,0 +1,177 @@
+import { handleActions, combineActions } from 'redux-actions';
+import { ethActions } from '../actions';
+
+const initialState = {
+  loading: false,
+  connecting: false,
+  wallet: null,
+  wallerError: null,
+  walletOptions: [],
+  tokenStakeContractInfoLoading: false,
+  tokenStakeContractInfo: null,
+  tokenStakeAddressInfo: {},
+  erc20Info: {},
+  erc20Balance: {},
+};
+
+const ethReducer = handleActions(
+  {
+    [ethActions.getErc20Balance.call]: (state, action) => ({
+      ...state,
+      erc20Info: {
+        ...state.erc20Info,
+        [`${action.payload.erc20Address}/${action.payload.account}`]: {
```

Comment:
If I m reading this correctly, its made to support multiple ERC20 tokens in the future?

Reply:
Precisely